api/test/wfl/integration/datarepo_test.clj

```diff
@@ -59,3 +61,28 @@
                  (datarepo/ingest-table dataset-id table-url table-name))]
             (is (= 1 row_count))
             (is (= 0 bad_row_count))))))))
+
+(deftest test-create-snapshot
+  ;; To test given a dataset, we can query BigQuery for row-ids
+  ;; and use them to create a snapshot!
+  (let [tdr-profile (env/getenv "WFL_TDR_DEFAULT_PROFILE")
+        definition "sarscov2-illumina-full-inputs.json"]
+    (testing "creating snapshots from dataset"
+      (fixtures/with-temporary-dataset
+        (datasets/unique-dataset-request tdr-profile definition)
+        (fn [dataset-id]
+          (let [{:keys [dataProject] :as dataset} (datarepo/dataset dataset-id)
+                table     "sarscov2_illumina_full_inputs"
+                today     (util/datetime->str (util/now))
+                yesterday (util/datetime->str (util/n-day-from-now -1))
+                row-ids   (->> (datarepo/compose-snapshot-query dataset table yesterday today)
+                               (bigquery/query-sync dataProject)
+                               (bigquery/flatten-rows)
+                               ; the fixture does not support cleaning up multiple snapshots yet
+                               (take 1))]
+            (fixtures/with-temporary-snapshot
+              (snapshots/unique-snapshot-request tdr-profile dataset table row-ids)
+              #(let [snapshot (datarepo/snapshot %)]
+                 (is (= % (:id snapshot)))))))))))
```

Comment:
This test is failing at this moment since the parent temp dataset looks like:
```
{:id "9331fab1-02ca-41d7-aed2-13fbd9c08c40",
 :name "sarscov2_illumina_full_inputs4edf44bf4c5145c682715881dc71ba7f",
 :description "COVID-19 sarscov2_illumina_full pipeline inputs",
 :defaultProfileId "XXX",
 :dataProject "broad-jade-dev-data",
 :defaultSnapshotId nil,
 :schema
 {:tables
  [{:name "sarscov2_illumina_full_inputs",
    :columns
    [{:name "flowcell_tgz", :datatype "fileref", :array_of false}
     {:name "reference_fasta", :datatype "fileref", :array_of false}
     {:name "amplicon_bed_prefix", :datatype "string", :array_of false}
     {:name "biosample_attributes",
      :datatype "fileref",
      :array_of true}
     {:name "instrument_model", :datatype "string", :array_of false}
     {:name "sra_title", :datatype "string", :array_of false}
     {:name "min_genome_bases", :datatype "int64", :array_of false}
     {:name "max_vadr_alerts", :datatype "int64", :array_of false}
     {:name "workspace_name", :datatype "string", :array_of false}
     {:name "terra_project", :datatype "string", :array_of false}
     {:name "extra", :datatype "string", :array_of false}],
    :primaryKey
    ["flowcell_tgz"
     "reference_fasta"
     "amplicon_bed_prefix"
     "instrument_model"
     "sra_title"],
    :partitionMode "date",
    :datePartitionOptions {:column "datarepo_ingest_date"},
    :intPartitionOptions nil,
    :rowCount nil}],
  :relationships [],
  :assets []},
 :createdDate "2021-03-11T05:09:25.655987Z"}
```
as you can see there are no rows in the dataset, thus no snapshot can be made. 

How to ingest rows into a created dataset (or, does it even make sense to create dataset during testing, rather than using a pre-defined testing dataset that has rows) is a question I'd like to get insights from @ehigham

Reply:
For this, I wouldnt worry about creating a temp dataset. I'd create a temp snapshot of this https://jade.datarepo-dev.broadinstitute.org/datasets/28dbedad-ca6b-4a4a-bd9a-b351b5be3617/query